src/Panel/Panel.d.ts

```diff
@@ -0,0 +1,37 @@
+import { PureComponent, ReactNode } from 'react';
+
+export namespace PanelProps {
+
+  type HeaderSize = 'tiny' | 'small' | 'medium';
+
+  type CollapseDirection = 'up' | 'left' | 'down' | 'right';
+
+  type Header = {
+    collapse?: {
+      direction: CollapseDirection,
+      onExpand: () => void,
+      onCollapse: () => void,
+      isCollapsed?: boolean
+    },
+    size?: HeaderSize,
+    content?: any,
+    title?: ReactNode | string, // TODO(typo): wtf
+    hideTitleWhenExpanded?: boolean,
+    menu?: any
+  }
+}
+
+export type PanelProps = {
+  children: ReactNode,
+  type: 'docked-top' | 'docked-left' | 'docked-bottom' | 'docked-bottom' | 'floating',
+  header?: PanelProps.Header,
+  loading?: boolean,
+  dark?: boolean,
+  softLoading?: boolean,
+  softLoadingDelay?: boolean,
```

Comment:
this should be a `number`
(to be precise, a `NonNegativeNumber` but we can't check that statically with Typescript)